pkg/scale/encode.go

```diff
@@ -351,6 +351,10 @@ func (es *encodeState) encodeUint(i uint) (err error) {
 
 // encodeUint128 encodes a Uint128
 func (es *encodeState) encodeUint128(i *Uint128) (err error) {
-	err = binary.Write(es, binary.LittleEndian, i.Bytes())
+	if i == nil {
+		err = fmt.Errorf("uint128 is nil: %v", i)
+		return
+	}
```

Comment:
thought this would be handled higher up - @timwu20 is this needed?

Reply:
It might not be. I added because when I was initially encoding with the old AccountInfo type there were cases where the fields weren't initialized and it was crashing. Shouldn't be an issue now since I changed that but I kept this anyways. Can remove if not necessary

Reply:
I don't think this should be necessary.  In most use cases when encoding, you would've have initialised a value for that pointer, instead of a zero case pointer.  Can you point me to the code that causes this panic?

Reply:
Yeah dot/rpc/modules/system_test.go where the AccountInfo type was used the struct fields weren't initiated and thats what caused my initial issue. This pr changed it so now they are initialized now